src/editor.rs

```diff
@@ -643,24 +643,32 @@ impl Editor {
     /// `forward` indicates whether to search forward or backward. Returns
     /// the row of a new match, or `None` if the search was unsuccessful.
     fn find(&mut self, query: &str, last_match: Option<usize>, forward: bool) -> Option<usize> {
-        // Number of rows to search
         let num_rows = if query.is_empty() { 0 } else { self.rows.len() };
-        let mut current = last_match.unwrap_or_else(|| num_rows.saturating_sub(1));
-        // TODO: Handle multiple matches per line
-        for _ in 0..num_rows {
-            current = (current + if forward { 1 } else { num_rows - 1 }) % num_rows;
-            let row = &mut self.rows[current];
-            if let Some(cx) = row.chars.windows(query.len()).position(|w| w == query.as_bytes()) {
-                // self.cursor.coff: Try to reset the column offset; if the match is after the
-                // offset, this will be updated in self.cursor.scroll() so that
-                // the result is visible
-                (self.cursor.x, self.cursor.y, self.cursor.coff) = (cx, current, 0);
-                let rx = row.cx2rx[cx];
-                row.match_segment = Some(rx..rx + query.len());
-                return Some(current);
+        let current = last_match.unwrap_or_else(|| num_rows.saturating_sub(1));
+        // 清除所有行的 match_segments
+        for row in &mut self.rows {
+            row.match_segments.clear();
+        }
+        let mut found_row = None;
+        for idx in 0..num_rows {
+            let i = (current + if forward { 1 } else { num_rows - 1 } + idx) % num_rows;
```

Comment:
I tried running the code, searching backwards (pressing up arrow key while the search prompt is open) seems to be broken now